poollet/volumepoollet/controllers/volume_controller.go

```diff
@@ -261,6 +261,27 @@ func (r *VolumeReconciler) prepareORIVolumeClass(ctx context.Context, volume *st
 	return class.Name, true, nil
 }
 
+func (r *VolumeReconciler) prepareORIVolumeEncryption(ctx context.Context, encryption *storagev1alpha1.VolumeEncryption) (*ori.EncryptionSpec, bool, error) {
+	if encryption == nil {
+		return nil, true, nil
+	}
+
+	encryptionSecret := &corev1.Secret{}
+	encryptionSecretKey := client.ObjectKey{Name: encryption.SecretRef.Name}
+	if err := r.Get(ctx, encryptionSecretKey, encryptionSecret); err != nil {
+		err = fmt.Errorf("error getting volume encryption secret %s: %w", encryptionSecretKey, err)
+		if !apierrors.IsNotFound(err) {
+			return nil, false, fmt.Errorf("error getting volume encryption secret %s: %w", encryption.SecretRef.Name, err)
+		}
+		return nil, false, nil
```

Comment:
What about an event on the volume for external users to know?

Reply:
Sounds good 👍